In [1]:
import scipy.stats as st
import numpy as np

Ej1: Una fabrica de azucar esta probando una máquina nueva que dice que distribuye azucar de 1Kg con menos de 2g de variación. En base a una muestra de 20 unidades se determinó una variación de 2.45g, ¿Con un 97% de confiabilidad se puede afirmar que el distribuidor de la máquina miente respecto a la precisión de la máquina? y con un 99% de confiabilidad?

In [24]:
sigma=2 # esla variacion de 2 grm / poblacionanl
s=2.45 # muestral
alpha=0.03
alpha2=0.01
n=20
M=s*(n-1)
E=M/sigma #estimador
k1=st.chi2.ppf(alpha,n-1)
k2=st.chi2.ppf(1-alpha,n-1) # k1 y k2 son los limites para los intervaalos de confianza
k1,E,k2

(np.float64(9.200440695386053),
 23.275000000000002,
 np.float64(32.15772037281206))

Podemos que al 97% de confiabilidad podemos aceptar que estadísticamente nuestra media sea alrededor de 2.

In [3]:
k1=st.chi2.ppf(alpha2,n-1)
k2=st.chi2.ppf(1-alpha2,n-1)
k1,E,k2

(np.float64(7.632729647571471),
 23.275000000000002,
 np.float64(36.19086912927004))

Al 99% de confiabilidad podemos tambien aceptar la hipótesis

In [4]:
k1=st.chi2.ppf(alpha/2,n-1)
k2=st.chi2.ppf(1-alpha/2,n-1)
M/k2,sigma**2,M/k1 # intervalo de confianza al 97% de confiabilidad para la varianza

(np.float64(1.3398784121617928), 4, np.float64(5.705468810560525))

In [5]:
k1=st.chi2.ppf(alpha2/2,n-1)
k2=st.chi2.ppf(1-alpha2/2,n-1)
M/k2,sigma**2,M/k1 # intervalo de confianza al 99% de confiabilidad para la varianza

(np.float64(1.2065131528458197), 4, np.float64(6.801606402189647))

Rem: Cuando hacemos una suma de variables aleatorias normales nos encontramos en una distribucion chi2 a n variables de libertad, donde el numero de variables de libertad es el número de sumandos de las leyes normales.

Ej2: 2 agencias de la misma empresa han calificado el desempeño de sus empleados del 1 al 100. La primera agencia obtuvo una media de 71.3 y una desviación estandar de 6.8 en base a una muestra de 12 empleados, la segunda agencia obtuvo una media de 65.1 con una desviación estandar 7.5 en base a una muestra de 15 empleados. ¿Se puede decir que ambas agencias tienen el mismo desempeño con un 94% de confiabilidad y al 98% de confiabilidad?

In [26]:
n1=12
n2=15
mu1=71.3
mu2=65.1
s1=6.8
s2=7.5
alpha=0.06
alpha2=0.02
E=np.sqrt((n1+n2-2)/(1/n1+1/n2))*np.abs(mu1-mu2)/np.sqrt((n1-1)*s1+(n2-1)*s2)
criterio=st.t.ppf(1-alpha/2,n1+n2-2)
E,criterio


(np.float64(5.969269964413238), np.float64(1.9700952169260397))

In [27]:
criterio=st.t.ppf(1-alpha2/2,n1+n2-2)
E,criterio

(np.float64(5.969269964413238), np.float64(2.4851071754106413))

Ej3: La ley de un  mineral es el porcentaje de mineral existente y explotable en medio de rocas de bajo valor. Para evaluar un yacimiento de plata se han extraido 30 muestras del mismo peso y se ha determinado que la ley media de los minerales analizados es de 2.5%, con una desviación estandar de 0.3%. Suponiendo que todo el yacimiento tiene la misma ley dar un intervalo de confianza al 95% de confiabilidad sobre la ley del mineral. Si el precio de la plata está a aproximadamente a $0.9\$/g$. Dar una estimación del valor del yacimiento a $1000 000\$ $ cerca si se estima que del yacimiento se puede extraer 5000 toneladas de roca

In [8]:
n=50
mu=5232
sigma=325
alpha=0.05
criterio=st.norm.ppf(1-alpha/2)
criterio2=st.t.ppf(1-alpha/2,n-1)
criterio,criterio2

(np.float64(1.959963984540054), np.float64(2.0095752371292397))

In [9]:
-criterio2*sigma/n+mu,criterio2*sigma/n+mu # intervalo de confianza de la media al 95% de confiabilidad

(np.float64(5218.93776095866), np.float64(5245.06223904134))

In [10]:
precio=0.9
Y=5000 *1000*1000*precio
Y

4500000000.0

In [11]:
st.t.ppf(alpha/2,n-1,mu,sigma/n),st.t.ppf(1-alpha/2,n-1,mu,sigma/n) # Sin pasar por determinar el criterio.

(np.float64(5218.93776095866), np.float64(5245.06223904134))

In [12]:
(12.15009104487176*2000, 12.489908955128241*2000)

(24300.182089743517, 24979.817910256483)

In [13]:
Y*2.479547703578673/100,Y*2.520452296421327/100 # intervalo de confianza del precio del yacimiento

(111579646.66104026, 113420353.33895974)

La estimación del precio del yacimiento está entre $(112M,113M)$

Ej4: Jugando calabozos y dragones para un movimiento debemos lanzar 4 dados de 8 caras. ¿Cuál es la probabilidad de obtener un 8 en la tirada?, ¿cuál es la probabilidad de tener al menos un par de 7's? y ¿cuál es la probabilidad de tener entre 2 y 3 números pares?

Veamos las posibilidades

* (2,2,2,2)
* (4,2,1,1)
* (3,2,2,1)

Por ley aditiva tenemos que considerar uno de los 3 casos

* Caso 1: Es 1 solo tener 2 en cada dado
* Caso 2: Es tener una permutacion de (4,2,1,1) es decir 4! posibilidades hay que dividir eso entre 2 lo que nos da 12 posibilidades
* Caso 3: Es tener una permutacion de (3,2,2,1) es decir 4! posibilidades que hay que divider entre 2 (otras 12 posibilidades)

Tenemos 25 casos favorables y P(8,4) casos posibles


In [14]:
import scipy as sci
import math as ma

In [15]:
25/ma.comb(8,4) #tenemos 35.71% probabilidades de obtener un 8 en la tirada

0.35714285714285715

Casos posibles es permutación 8,2

In [16]:
ma.comb(8,2)/ma.comb(8,4) #tenemos 40% de probabilidades de obtener al menos un par de 7 en la tirada

0.4

Tener 2 numero pares implica Combinatoria (4,2) y otros 2 numeros impares implica lo mismo

In [17]:
(ma.comb(4,2)*ma.comb(4,2)+ma.comb(4,3)*4)/ma.comb(8,4) # tenemos 74.29% de probabilidades de tener entre 2 y 3 numeros pares

0.7428571428571429

Ej: El tiempo promedio de recepción de nuevas solicitudes de una página web es 1 minuto. ¿Cuál es la probabilidad de tener una nueva solicitud en menos de 90 segundos?, ¿cuál es la probabilidad de recibir una nueva solicitud entre 2 a 3 minutos?, ¿Cuál es la probabilidad de recibirla después de 5 minutos?

Vamos a utilizar la distribución exponencial

In [18]:
st.expon.cdf(1.5,1) #tenemos 39.35%  de probabilidades de recibirla en los primeros 90 segundos

np.float64(0.3934693402873666)

In [19]:
st.expon.cdf(3,1)-st.expon.cdf(2,1) #tenemos 23.25% de probabilidades de recibirla entre los minutos 2 y 3

np.float64(0.23254415793482963)

In [20]:
1-st.expon.cdf(5,1) # tenemos 1.83% de probabilidades de recibirla despues de los 5 minutos

np.float64(0.01831563888873422)

Ej6: 2 paralelos de la misma materia dieron el mismo examen con medias de 65 y 73 puntos además de desviaciones estandar de 5 y 7 puntos. ¿Qué paralelo tiene mayor probabilidad de haber aprobado el curso (nota de aprobación 60)?, ¿Qué curso tiene mayor probabilidad de obtener menos de 75 puntos?

In [21]:
# Primer curso aprobación
1-st.norm.cdf(60,65,5) # tenemos 84.13% de probabilidades de haber aprobado


np.float64(0.8413447460685429)

In [22]:
# Segundo curso aprobación
1-st.norm.cdf(60,73,7) # tenemos 96.84% de probabilidades de haber aprobado


np.float64(0.9683545838833274)

El segundo curso tiene más probabilidades de haber aprobado

In [23]:
st.norm.cdf(75,65,5), st.norm.cdf(75,73,7)

(np.float64(0.9772498680518208), np.float64(0.6124515189020077))

En el primer caso tenemos $97.72\%$ de probabilidades de haber obtenido menos de 75 puntos y en el segundo caso $61.25\%$, por lo cuál es más probable que el primer curso tenga menos que 75  